floss/language/go/extract.py

```diff
@@ -0,0 +1,408 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import pathlib
+import argparse
+from typing import List, Iterable, Optional
+from itertools import chain
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 6
+
+
+def extract_strings_from_import_data(pe: pefile.PE) -> Iterable[StaticString]:
+    """Extract strings from the import data"""
+
+    for entry in pe.DIRECTORY_ENTRY_IMPORT:
+        for imp in entry.imports:
+            if imp.name is not None:
+                yield StaticString(string=imp.name.decode("utf-8"), offset=imp.address, encoding=StringEncoding.ASCII)
+
+
+def extract_build_id(pe: pefile.PE, section_data, section_va) -> Iterable[StaticString]:
+    # Build ID is a string that starts with "\xff\x20 Go build ID: " and ends with "\n"
+    # FF 20 47 6F 20 62 75 69  6C 64 20 49 44 3A 20 22  . Go build ID: "
+    # 36 4E 31 4D 77 6E 30 31  72 46 6E 41 51 4B 62 5A  6N1Mwn01rFnAQKbZ
+    # 73 46 5A 32 2F 38 41 6B  75 63 4B 46 58 4D 49 54  sFZ2/8AkucKFXMIT
+    # 63 51 52 49 75 55 5F 79  32 2F 62 76 4F 67 56 37  cQRIuU_y2/bvOgV7
+    # 52 4D 54 72 77 73 7A 5A  39 57 7A 69 6C 64 2F 72  RMTrwszZ9Wzild/r
+    # 33 31 50 47 70 61 6B 2D  48 77 36 4B 72 77 59 6E  31PGpak-Hw6KrwYn
+    # 52 4E 73 22 0A 20 FF CC  CC CC CC CC CC CC CC CC  RNs". ..........
+
+    build_id_regex = re.compile(b"(?<=\xff\x20)(.)*\x0A")
+
+    for m in build_id_regex.finditer(section_data):
+        addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
```

Comment:
do you think we should use the raw offset or the virtual address?
our other static strings use the raw file offset

Reply:
I'm in favor of raw offset and think we can reduce the number of arguments needed for these functions